go/vt/sqlparser/sql.y

```diff
@@ -433,13 +439,13 @@ select_expression:
   {
     $$ = &NonStarExpr{Expr: $1, As: $2}
   }
-| table_id '.' '*'
+| sql_id '.' '*'
```

Comment:
Actually I found a cleaner way around this, as you can now see in the latest push